ccmlib/node.py

```diff
@@ -551,6 +551,12 @@ def watch_log_for(self, exprs, from_mark=None, timeout=600, process=None, verbos
                     if start + timeout < time.time():
                         raise TimeoutError(time.strftime("%d %b %Y %H:%M:%S", time.gmtime()) + " [" + self.name + "] Missing: " + str([e.pattern for e in tofind]) + ":\n" + reads[:50] + ".....\nSee {} for remainder".format(filename))
 
+                    # update status because meanwhile node could have died....
+                    pid_before_check = self.pid
+                    self.__update_status()
+                    if not self.is_running():
+                        raise RuntimeError("node {n} with pid {pid} is not running".format(n=self.name, pid=pid_before_check))
```

Comment:
I wonder if it can cause problems... in the `node.stop()`
it should not but there is a possible race between reading log and the moment pid disappears